recipes-bsp/uefi/edk2-hikey960_git.bb

```diff
@@ -0,0 +1,88 @@
+require edk2_git.bb
+
+COMPATIBLE_MACHINE = "hikey960"
+
+DEPENDS_append = " dosfstools-native mtools-native fakeroot-native grub"
+
+inherit deploy pythonnative
+
+SRCREV_edk2 = "7efa39f3631ca8a42cc17210fb293b850aecbf3f"
+SRCREV_atf = "645904c8106640c6f694ce12d26ef4cb4184e196"
+SRCREV_openplatformpkg = "0e34c233f9d61c5dafa300176b2b022745d30d37"
+SRCREV_uefitools = "9857704d962ed9dad02e7cee2f3de240d132c872"
+SRCREV_lloader = "2835202b47bf734e48c6c25cd35ffbeecb8aa141"
+SRCREV_toolsimageshikey960 = "5716ead2fbf063b1ff578de7d6004a100d2ff92d"
+
+SRC_URI = "git://github.com/96boards-hikey/edk2.git;name=edk2;branch=testing/hikey960_v2.5 \
+           git://github.com/96boards-hikey/arm-trusted-firmware.git;name=atf;branch=testing/hikey960_v1.1;destsuffix=git/atf \
+           git://github.com/96boards-hikey/OpenPlatformPkg.git;name=openplatformpkg;branch=testing/hikey960_v1.3.4;destsuffix=git/OpenPlatformPkg \
+           git://github.com/96boards-hikey/uefi-tools.git;name=uefitools;branch=testing/hikey960_v1;destsuffix=git/uefi-tools \
+           git://github.com/96boards-hikey/l-loader.git;name=lloader;branch=testing/hikey960_v1.2;destsuffix=git/l-loader \
+           git://github.com/96boards-hikey/tools-images-hikey960.git;name=toolsimageshikey960;destsuffix=git/tools-images-hikey960 \
+           file://grub.cfg.in \
+           file://config \
+          "
+
+# /usr/lib/edk2/bl1.bin not shipped files. [installed-vs-shipped]
+INSANE_SKIP_${PN} += "installed-vs-shipped"
+
+# workaround EDK2 is confused by the long path used during the build
+# and truncate files name expected by VfrCompile
+do_patch[postfuncs] += "set_max_path"
+set_max_path () {
+    sed -i -e 's/^#define MAX_PATH.*/#define MAX_PATH 511/' ${S}/BaseTools/Source/C/VfrCompile/EfiVfr.h
```

Comment:
No, what will help is to report a bug against VFrCompile and fix it upstream. Alternatively, you should change your build path to something shorter like /srv/oe. There's several corner cases that we haven't tested and limiting the changes is my preferred option instead of increasing to a higher value without understanding the consequences.